source/Connection/AMDSServer.cpp

```diff
@@ -0,0 +1,90 @@
+#include "AMDSServer.h"
+
+#include <QtNetwork>
+
+#include "source/Connection/AMDSClientTCPSocket.h"
+#include "source/ClientRequest/AMDSClientRequest.h"
+#include "source/ClientRequest/AMDSClientIntrospectionRequest.h"
+
+AMDSServer::AMDSServer(QString hostName, quint16 portNumber, QObject *parent) :
+    QObject(parent)
+{
+	hostName_ = hostName;
+	portNumber_ = portNumber;
+}
+
+AMDSServer::~AMDSServer()
+{
+	// release the resources of the active sockets
+	foreach (QString key,  activeTCPSockets_.keys()) {
+		removeTCPSocket(activeTCPSockets_.value(key));
+	}
+}
+
+void AMDSServer::setBufferNames(QStringList activeBufferNames) {
+	bufferNames_.clear();
+	bufferNames_.append(activeBufferNames);
+}
+
+AMDSClientTCPSocket * AMDSServer::establishSocketConnection()
+{
+	AMDSClientTCPSocket * clientTCPSocket = new AMDSClientTCPSocket(hostName_, portNumber_);
```

Comment:
I'm pretty sure I've asked this and remember being satisfied with the answer, but just so my mind is at ease, is the memory of the created sockets here cleaned up in requested classes?